EIPS/eip-6066.md

```diff
@@ -0,0 +1,127 @@
+---
+eip: 6066
+title: Signature Validation Method for NFTs
+description: A way to verify signatures when the signing entity is an EIP-721 or EIP-1155 NFT
+author: Jack Boyuan Xu (@boyuanx)
+discussions-to: https://ethereum-magicians.org/t/eip-6066-signature-validation-method-for-nfts/
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-11-29
+requires: 721, 1155, 1271, 5750
+---
+
+## Abstract
+
+While **E**xternally **O**wned **A**ccounts can validate signed messages with `ecrecover()` and smart contracts can validate signatures using specifications outlined in [EIP-1271](./eip-1271.md), currently there is no standard method to create or validate signatures made by NFTs. We propose a standard way for anyone to validate whether a signature made by an NFT is valid. This is possible via a modified signature validation function originally found in [EIP-1271](./eip-1271.md): `isValidSignature(tokenId, hash, data)`.
+
+## Motivation
+
+With billions of ETH in trading volume, the **N**on-**F**ungible **T**oken standard has exploded into tremendous popularity in recent years. Despite the far-reaching implications of having unique tokenized items on-chain, NFTs have mainly been used to represent artwork in the form of avatars or profile pictures. While this is certainly not a trivial use case for the [EIP-721](./eip-721.md) & [EIP-1155](./eip-1155.md) token standards, we reckon more can be done to aid the community in discovering alternative uses for NFTs.
```

Comment:
This section doesn't convince me, a skeptical reader, that this EIP is necessary. What use cases require a signature from the NFT itself, and not its current holder? What makes this proposal different from other signature schemes?

Reply:
This is most useful when the signing entity is a certain position within an organization or, broadly speaking, any identity construct that is not be tied to any singular address over time. It allows signatures to act as a stamp of approval representing the consent of an organization instead of any individual person. Currently, all consent represented by signatures is strictly tied to a single address that's non-transferrable. This EIP adds a layer of abstraction and provides flexibility when consent transfer is needed.